compiler/compile/OMRCompilation.hpp

```diff
@@ -168,6 +172,64 @@ typedef TR::SparseBitVector SharedSparseBitVector;
    #define addDebug(string)   ((void)0)
 #endif
 
+/**
+ * \def performTransformation(comp, msg, ...) 
+ *
+ * \brief Macro that is used to guard optional steps in compilation in order to
+ *        aid debuggability  
+ *
+ *
+ * Ostensibly, the purpose of performTransformation is to allow code
+ * transformations to be selectively disabled during debugging in order to
+ * isolate a buggy optimization. But this description fails to capture the
+ * important effect that performTransformation has on the maintainability of
+ * Testarossa’s code base.
+ * 
+ * Calls to performTransformation can (and should) be placed around any part of
+ * the code that is optional; in an optimizer, that’s a lot of code. Tons of
+ * Testarossa code is there only to improve performance–not for correctness–and
+ * can therefore be guarded by performTransformation.
+ * 
+ * A call in a hypothetical dead code elimination might look like this:
+ * 
+ *     if (performTransformation2c(comp(),
+ *           "%sDeleting dead candidate [%p]\n", OPT_DETAILS, candidate->_node))
+ *        {
+ *        // Logic to delete some dead code. 
+ *        }
+ * 
+ * When this opt runs on a method that is being logged, all calls to
+ * performTransformation will be assigned a unique number. Running the test
+ * case with lastOptTransformationIndex=n will prevent subsequent
+ * transformations from occurring. By adjusting n and observing when the test
+ * passes and fails, it is possible to narrow down the failing transformation
+ * using a binary-search approach.
+ * 
+ * But beyond just allowing this opt to be enabled and disabled, this idiom
+ * also does the following:
+ * 
+ *  -  It describes what the code does without needing a comment
+ *  -  It reports the transformation in the log
+ *  -  It gives people unfamiliar with the opt a way to locate the code that
+ *     caused a particular transformation
+ *  -  Combined with countOptTransformations, it allows you to locate methods
+ *     in which this opt is occurring
+ * 
+ * Most importantly, it identifies exactly the code that should be skipped if
+ * someone wanted to prevent this opt from occurring in certain cases. For
```

Comment:
Hm.... Escape Analysis?  I'm not familiar with that since its not in the OMR code base.  :-)

Please delete the whole "For instance," sentence.